treeherder/model/models.py

```diff
@@ -706,6 +704,8 @@ class Meta:
             # this is intended to speed up queries for specific platform /
             # option collections on a push
             ('machine_platform', 'option_collection_hash', 'push'),
+            # speed up cycle data
+            ('repository', 'submit_time'),
```

Comment:
I think this should be a standard index on only `submit_time`?
Whilst the current query breaks things down by repository, I don't think we should do that longer term. (I'm not saying that we should change that behaviour now, but a single index would fill both use-cases.)